crates/ruff/src/rules/flake8_bugbear/rules/useless_expression.rs

```diff
@@ -1,41 +1,77 @@
-use rustpython_parser::ast::{Constant, ExprKind, Stmt, StmtKind};
+use rustpython_parser::ast::{Constant, Expr, ExprKind};
 
 use ruff_diagnostics::{Diagnostic, Violation};
 use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::helpers::contains_effect;
 use ruff_python_ast::types::Range;
 
 use crate::checkers::ast::Checker;
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum Kind {
+    Expression,
+    Attribute,
+}
+
 #[violation]
-pub struct UselessExpression;
+pub struct UselessExpression {
+    kind: Kind,
+}
 
 impl Violation for UselessExpression {
     #[derive_message_formats]
     fn message(&self) -> String {
-        format!("Found useless expression. Either assign it to a variable or remove it.")
+        match self.kind {
+            Kind::Expression => {
+                format!("Found useless expression. Either assign it to a variable or remove it.")
+            }
+            Kind::Attribute => {
+                format!(
+                    "Found useless attribute access. Either assign it to a variable or remove it."
+                )
+            }
+        }
     }
 }
 
 /// B018
-pub fn useless_expression(checker: &mut Checker, body: &[Stmt]) {
-    for stmt in body {
-        if let StmtKind::Expr { value } = &stmt.node {
-            match &value.node {
-                ExprKind::List { .. } | ExprKind::Dict { .. } | ExprKind::Set { .. } => {
-                    checker
-                        .diagnostics
-                        .push(Diagnostic::new(UselessExpression, Range::from(value)));
-                }
-                ExprKind::Constant { value: val, .. } => match &val {
-                    Constant::Str { .. } | Constant::Ellipsis => {}
-                    _ => {
-                        checker
-                            .diagnostics
-                            .push(Diagnostic::new(UselessExpression, Range::from(value)));
-                    }
-                },
-                _ => {}
+pub fn useless_expression(checker: &mut Checker, value: &Expr) {
+    // Ignore comparisons, as they're handled by `useless_comparison`.
+    if matches!(value.node, ExprKind::Compare { .. }) {
+        return;
+    }
+
+    // Ignore strings, to avoid false positives with docstrings.
```

Comment:
Adding a small code snipped here of a case where a docstring causes a false positive would be helpful. 

Is there a way to determine whether a string is a docstring?